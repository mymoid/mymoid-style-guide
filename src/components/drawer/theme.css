@import '../../colors.css';
@import '../../variables.css';
@import '../../media.css';
@import '../app_bar/config.css';
@import './config.css';

.wrapper {
  position: relative;
  z-index: var(--z-index-zero);
}

.drawer {
  background-color: var(--drawer-background-color);
  box-shadow: var(--shadow-2p);
  color: var(--drawer-text-color);
  display: block;
  height: 100vh;
  /*overflow-x: hidden;*/
  /*overflow-y: auto;*/
  pointer-events: none;
  position: fixed;
  top: var(--appbar-height); /*TODO: top: 0, */
  transform-style: preserve-3d;
  transition: transform var(--animation-duration) var(--animation-curve-default);
  transition-delay: 0s;
  width: var(--drawer-mobile-width);
  will-change: transform;

  @apply --reset;

  &.active {
    pointer-events: all;
    transform: translateX(0px);
    transition-delay: var(--animation-delay);
  }

  &.right {
    right: 0;

    &:not(.active) {
      pointer-events: all; /*TODO: remove this, may be need an collapse state  */
      transform: translateX(100%);
    }
  }

  &.left {
    left: 0;

    &:not(.active) {
      pointer-events: all; /*TODO: remove this, may be need an collapse state  */
      transform: translateX(
        calc(calc(-1 * 100%) + 60px)
      ); /*TODO: override in dashboard.css*/
    }
  }
}

@media screen and (--larger-than-xs-viewport) {
  .drawer {
    width: var(--drawer-desktop-width);
  }
}
