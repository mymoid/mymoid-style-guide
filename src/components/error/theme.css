@import '../../colors.css';
@import '../../variables.css';
@import './config.css';

.error {
  @apply --reset;

  font-family: Lato;
  font-size: var(--error-font-size);
  color: var(--error-color);
  letter-spacing: 0;
  padding: var(--error-padding);
}
