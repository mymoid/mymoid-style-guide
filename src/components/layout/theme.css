@import '../../colors.css';
@import '../../variables.css';
@import '../../media.css';
@import './config.css';

@import '../app_bar/config.css';
@import '../drawer/config.css';

:root {
  --layout-side-transition: all var(--animation-duration)
    var(--animation-curve-default);
}

.layout {
  align-items: stretch;
  display: flex;
  flex: 1;
  flex-direction: column;
  justify-content: space-between;
  min-height: 100vh;
  min-width: 100%;
  position: relative;

  @apply --reset;
}

.panel {
  left: 0;
  position: absolute;
  right: 0;
  top: 0;

  &:not(.bodyScroll) {
    height: 100vh;
    max-height: 100vh;
    overflow-y: scroll;
  }
}

.navDrawerDrawer {
  z-index: var(--z-index-high);

  &.pinned {
    box-shadow: none;
    left: 0;
  }

  &.clipped {
    height: calc(100vh - var(--appbar-height));
    padding-top: calc(0.5 * var(--unit));
    top: var(--appbar-height);

    @media screen and (--xxs-viewport) and (--portrait) {
      height: calc(100vh - var(--appbar-height-m-portrait));
      top: var(--appbar-height-m-portrait);
    }

    @media screen and (--xs-viewport) and (--landscape) {
      height: calc(100vh - var(--appbar-height-m-landscape));
      top: var(--appbar-height-m-landscape);
    }
  }
}

.appbarInner,
.panel {
  transition: var(--layout-side-transition);
}

.appbarFixed {
  &.appbarAppBar {
    z-index: var(--z-index-high);
  }

  & .panel {
    top: var(--appbar-height);

    &:not(.bodyScroll) {
      height: calc(100vh - var(--appbar-height));
      max-height: calc(100vh - var(--appbar-height));
      overflow-y: scroll;
    }

    @media screen and (--xxs-viewport) and (--portrait) {
      top: var(--appbar-height-m-portrait);
    }

    @media screen and (--xs-viewport) and (--landscape) {
      top: var(--appbar-height-m-landscape);
    }
  }
}

.navDrawerPinned {
  & .appbarLeftIcon {
    display: none;
  }

  & .panel {
    left: var(--drawer-mobile-width);
  }

  &:not(.navDrawerClipped) {
    & .appbarAppBar {
      padding-left: calc(var(--drawer-mobile-width) + var(--appbar-h-padding));
    }
  }

  @media screen and (--larger-than-xs-viewport) {
    & .panel {
      left: var(--drawer-desktop-width);
    }

    &:not(.navDrawerClipped) {
      & .appbarAppBar {
        padding-left: calc(
          var(--drawer-desktop-width) + var(--appbar-h-padding)
        );
      }
    }
  }
}

.navDrawerClipped {
  & .navDrawerWrapper {
    position: relative;
    z-index: var(--z-index-normal);
  }
}

.panelFloatingCentered {
  margin-left: auto;
  position: relative;
  margin-right: auto;
  margin-top: calc(0.35 * var(--floating-panel-height));
  margin-bottom: calc(0.35 * var(--floating-panel-height));
}

.panelContent {
  background: var(--panel-content-background-color);
  box-shadow: 0 5px 40px 0 rgba(0, 0, 0, 0.31);
  border-radius: calc(0.1 * var(--unit));
}

.panelActions {
  align-items: center;
  display: flex;
  justify-content: center;
  padding: var(--panel-padding-sm);
  background: var(--panel-actions-background);
  opacity: var(--panel-actions-background-opacity);
  box-shadow: var(--panel-actions-box-shadow);

  & [data-mymoid='button'] {
    margin: 0 calc(var(--panel-padding-sm) / 2);
    min-width: 0;
    padding: 0 var(--panel-padding-sm);

    &:first-child {
      margin-left: 0;
    }

    &:last-child {
      margin-right: 0;
    }
  }
}

.panelTitle {
  align-items: center;
  display: flex;
  justify-content: center;
  padding: var(--panel-title-padding-sm);
}

@define-mixin sidebarIncrements $platform {
  & .sidebarDrawer {
    width: calc(var(--standard-increment-$(platform)) * $(increment));
  }

  &.sidebarPinned {
    & .panel {
      right: calc(var(--standard-increment-$(platform)) * $(increment));
    }

    &:not(.sidebarClipped) {
      & .appbarAppBar {
        padding-right: calc(
          var(--standard-increment-$(platform)) * $(increment) +
            var(--appbar-h-padding)
        );
      }
    }
  }
}

@each $increment in 1, 2, 3, 4, 5, 6, 7, 8, 9, 10, 11, 12 {
  .sidebarWidth$(increment) {
    & .sidebarDrawer {
      width: 100%;
    }

    @media screen and (--larger-than-sm-viewport) {
      @mixin sidebarIncrements desktop;
    }
  }
}

@each $increment in 1, 2, 3, 4, 5, 6, 7, 8, 9 {
  .sidebarWidth$(increment) {
    @media screen and (--larger-than-xs-viewport) and (--landscape) {
      @mixin sidebarIncrements mobile;
    }

    @media screen and (--larger-than-xs-viewport) and (orientation: portrait) {
      @mixin sidebarIncrements desktop;
    }
  }
}

@each $increment in 1, 2, 3, 4, 5, 6, 7, 8, 9, 10 {
  .sidebarWidth$(increment) {
    @media screen and (--larger-than-sm-tablet-viewport) {
      @mixin sidebarIncrements desktop;
    }
  }
}

@each $pct in 25, 33, 50, 66, 75, 100 {
  .sidebarWidth$(pct) {
    & .sidebarDrawer {
      width: 100%;
    }
  }
}

@media screen and (--larger-than-sm-tablet-viewport) {
  @each $pct in 25, 33, 50, 66, 75, 100 {
    .sidebarWidth$(pct) {
      & .panel {
        right: calc($(pct) * 1%);
      }

      & .sidebarDrawer {
        width: calc($(pct) * 1%);
      }

      &:not(.sidebarClipped) {
        & .appbarAppBar {
          padding-right: calc(calc($(pct) * 1%) + var(--appbar-h-padding));
        }
      }
    }
  }
}
