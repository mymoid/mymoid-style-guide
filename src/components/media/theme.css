@import '../../colors.css';
@import '../../variables.css';
@import './config.css';

.media {
  background-position: center center;
  background-repeat: no-repeat;
  background-size: cover;
  position: relative;

  &.wide,
  &.square {
    width: 100%;

    & .content {
      height: 100%;
      position: absolute;
    }

    & .content > iframe,
    & .content > video,
    & .content > img {
      max-width: 100%;
    }
  }

  &::after {
    content: '';
    display: block;
    height: 0;
  }

  &.wide::after {
    padding-top: 56.25%;
  }

  &.square::after {
    padding-top: 100%;
  }

  & .content {
    display: flex;
    flex-direction: column;
    justify-content: flex-end;
    left: 0;
    overflow: hidden;
    position: relative;
    top: 0;
    width: 100%;
  }
}
