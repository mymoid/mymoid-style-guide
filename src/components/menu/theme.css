@import '../../colors.css';
@import '../../variables.css';
@import './config.css';

.iconMenu {
  display: inline-block;
  position: relative;
  text-align: center;

  @apply --reset;

  & .icon {
    cursor: pointer;
    /*TODO: refactor magic number to na*/
    height: 100%;
  }
}

.active {
  background-color: transparent;
}

.menu {
  display: inline-block;
  position: relative;

  &.topLeft {
    position: absolute;
    left: 0;
    top: 0;

    & > .outline {
      transform-origin: 0 0;
    }
  }

  &.topRight {
    position: absolute;
    left: 100%;
    top: 10px;

    & > .outline {
      transform-origin: 100% 0;
    }
  }

  &.bottomLeft {
    position: absolute;
    top: calc(100% - 1px);
    left: 0;

    & > .outline {
      transform-origin: 0 100%;
    }
  }

  &.bottomRight {
    position: absolute;
    /*TODO: refactor this*/
    top: calc(100% - 1px);
    right: 0;

    & > .outline {
      transform-origin: 100% 100%;
    }
  }

  &:not(.static) {
    pointer-events: none;
    z-index: var(--z-index-higher);

    & > .outline {
      opacity: 0;
      transform: scale(0);
      transition: transform var(--menu-expand-duration)
          var(--animation-curve-default),
        opacity var(--menu-fade-duration) var(--animation-curve-default);
      will-change: transform;
    }

    & > .menuInner {
      left: 0;
      margin: 0;
      opacity: 0;
      position: absolute;
      top: 0;
    }

    &.rippled:not(.active) {
      & > .outline {
        transition-delay: var(--menu-ripple-delay);
      }

      & > .menuInner {
        transition-delay: var(--menu-ripple-delay);
      }
    }

    &.active {
      pointer-events: all;

      & > .outline {
        opacity: 1;
        transform: scale(1);
      }

      & > .menuInner {
        opacity: 1;
        transition: opacity var(--menu-fade-duration)
            var(--animation-curve-default),
          clip var(--menu-expand-duration) var(--animation-curve-default);
      }
    }
  }
}

.outline {
  background-color: var(--menu-background-color);
  border-radius: var(--menu-outline-border-radius);
  box-shadow: var(--shadow-2p);
  display: block;
  left: 0;
  position: absolute;
  top: 0;
}

.menuInner {
  display: block;
  list-style: none;
  padding: var(--menu-padding);
  position: relative;
  text-align: left;
  white-space: nowrap;
}

.menuItem {
  align-items: center;
  color: var(--menu-item-color);
  display: flex;
  font-size: var(--menu-item-font-size);
  height: var(--menu-item-height);
  overflow: hidden;
  padding: 0 var(--menu-item-padding);
  position: relative;

  &:not(.disabled):hover {
    background-color: var(--menu-item-hover-background);
    cursor: pointer;
  }

  &.disabled {
    opacity: 0.5;
    pointer-events: none;
  }

  &.selected {
    background-color: var(--menu-item-selected-background);
    font-weight: 500;
  }

  & .ripple {
    color: var(--color-white); /*TODO: --color-text-secondary*/
  }

  & .icon {
    font-size: var(--menu-item-icon-font-size);
    width: var(--menu-item-icon-size);
    height: initial;
    font-family: Lato;
    color: #cdd8dd;
    letter-spacing: 0;
  }
}

.caption {
  flex-grow: 1;
  font-size: var(--font-size-normal);
}

.shortcut {
  margin-left: var(--menu-item-padding);
}

.menuDivider {
  background-color: var(--color-divider);
  border: 0;
  display: block;
  height: 1px;
  margin: var(--menu-divider-height) 0;
  outline: 0;
  padding: 0;
  width: 100%;
}

.profileTitle {
  align-items: center;
  display: flex;
  text-transform: none;
  text-align: left;

  & [data-mymoid='avatar'] {
    margin-right: calc(2.1 * var(--unit));
  }

  & .title {
    font-family: Lato;
    font-size: calc(2 * var(--unit));
    font-weight: 500;
    letter-spacing: 0.02em;
    line-height: 1;
    margin: 0;
    padding: 0;
  }

  & .subtitle {
    font-family: Lato;
    font-weight: 300;
    color: var(--menu-profile-subtitle-color);
    font-size: calc(1.2 * var(--unit));
    text-transform: uppercase;
    font-weight: 400;
    letter-spacing: 0;
    line-height: calc(2.3 * var(--unit));
    margin: 0;
    padding: 0;

    &.active {
      color: var(--menu-profile-subtitle-active-color);
    }
  }

  &.large {
    padding: var(--menu-profile-title-padding-lg)
      var(--menu-profile-title-padding)
      calc(var(--menu-profile-title-padding) - 0.2 * var(--unit));

    & .title {
      font-size: calc(2.4 * var(--unit));
      -moz-osx-font-smoothing: grayscale;
      font-weight: 400;
      line-height: 1.25;
    }
  }

  &.small {
    padding: var(--menu-profile-title-padding) 6px;

    & .title {
      letter-spacing: 0;
      font-size: calc(1.6 * var(--unit));
      line-height: 1.4;
      color: var(--menu-profile-title-color);

      &.active {
        color: var(--menu-profile-title-active-color);
      }
    }

    & .subtitle {
      letter-spacing: calc(0.07 * var(--unit));
      font-weight: 500;
      line-height: 2.3;
    }
  }
}
