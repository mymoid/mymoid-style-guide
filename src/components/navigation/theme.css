@import '../../colors.css';
@import '../../variables.css';
@import './config.css';

.horizontal {
  & > [data-mymoid='button'],
  & > [data-mymoid='link'] {
    display: inline-block;
    margin: 0 calc(var(--navigation-space) / 2);
  }
}

.vertical {
  & > [data-mymoid='button'],
  & > [data-mymoid='link'] {
    display: block;
    margin: calc(var(--navigation-space) / 2);
  }
}

.vertical,
.horizontal {
  padding: calc(var(--navigation-space) / 2);

  @apply --reset;

  & > [data-mymoid='link'] {
    color: var(--navigation-color);
  }
}
