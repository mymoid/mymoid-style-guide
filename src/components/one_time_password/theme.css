@import '../../colors.css';
@import '../../variables.css';
@import './config.css';

.codeInput {
  & > *:focus {
    outline: none;
  }

  & > *::-webkit-input-placeholder {
    font-family: Lato;
    font-weight: 300;
    font-size: var(--otp-input-font-size);
    color: var(--otp-input-placeholder-color);
    letter-spacing: 8px;
  }

  input[type='number']::-webkit-outer-spin-button,
  input[type='number']::-webkit-inner-spin-button {
    -webkit-appearance: none;
    margin: 0;
  }
  input[type='text']::-webkit-input-placeholder {
    font-family: Lato;
    font-weight: 300;
    font-size: 48px;
    color: #d8d8d8;
    letter-spacing: 8px;
  }
  input[type='number'] {
    -moz-appearance: textfield;
  }
}
