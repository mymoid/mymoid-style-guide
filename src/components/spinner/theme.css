@import '../../colors.css';
@import '../../variables.css';
@import './config.css';

@keyframes dots-pulse {
  0% {
    opacity: 0.1;
  }
  15% {
    opacity: 1;
  }
  100% {
    opacity: 0.1;
  }
}

.circle {
  border-radius: 100%;
  display: inline-block;
  margin-right: calc(0.3 * var(--unit));
  background-color: var(--dots-spinner-color);
  width: calc(0.8 * var(--unit));
  height: calc(0.8 * var(--unit));
  animation-name: dots-pulse;
  animation-duration: inherit;
  animation-timing-function: ease-in-out;
  animation-iteration-count: infinite;
  &:last-child {
    margin-right: 0;
  }
}
