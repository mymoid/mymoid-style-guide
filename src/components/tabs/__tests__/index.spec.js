import React, { Component } from 'react';
import Adapter from 'enzyme-adapter-react-16';
import Enzyme, { mount, shallow } from 'enzyme';
import { mountToJson } from 'enzyme-to-json';
import { Tabs } from '../Tabs';
import { Tab } from '../Tab';
import { TabContent } from '../TabContent';
import theme from '../theme.css';

Enzyme.configure({ adapter: new Adapter() });

describe('Tabs', () => {
  class Composition extends Component {
    constructor() {
      super();
      this.state = { index: 0 };
    }

    render() {
      return (
        <Tabs
          theme={{ pointer: 'pointer-class' }}
          index={this.state.index}
          {...this.props}
        >
          <Tab label="tab1">tab1</Tab>
          <Tab label="tab2">tab2</Tab>
        </Tabs>
      );
    }
  }

  it('defaults to only rendering the current tab', () => {
    const wrapper = mount(<Composition />);
    expect(mountToJson(wrapper)).toMatchSnapshot();
    // expect(wrapper.find(TabContent).length).toEqual(1);
    // expect(
    //   wrapper
    //     .find(TabContent)
    //     .first()
    //     .prop('tabIndex')
    // ).toEqual(0);
    //
    // wrapper.instance().setState({ index: 1 });
    // expect(wrapper.find(TabContent).length).toEqual(1);
    // expect(
    //   wrapper
    //     .find(TabContent)
    //     .first()
    //     .prop('tabIndex')
    // ).toEqual(1);
    // expect(mountToJson(wrapper)).toMatchSnapshot();
  });

  it('renders inactive tabs when hideMode is set to display', () => {
    const wrapper = mount(<Composition hideMode="display" />);
    expect(mountToJson(wrapper)).toMatchSnapshot();
    // expect(wrapper.find(TabContent).length).toEqual(2);
    // expect(
    //   wrapper
    //     .find(TabContent)
    //     .at(0)
    //     .prop('hidden')
    // ).toEqual(false);
    // expect(
    //   wrapper
    //     .find(TabContent)
    //     .at(1)
    //     .prop('hidden')
    // ).toEqual(true);
    //
    // wrapper.instance().setState({ index: 1 });
    // expect(wrapper.find(TabContent).length).toEqual(2);
    // expect(
    //   wrapper
    //     .find(TabContent)
    //     .at(0)
    //     .prop('hidden')
    // ).toEqual(true);
    // expect(
    //   wrapper
    //     .find(TabContent)
    //     .at(1)
    //     .prop('hidden')
    // ).toEqual(false);
  });

  describe('#render', () => {
    it('does not use fixed by default', () => {
      const wrapper = shallow(<Tabs theme={theme} />);
      expect(
        wrapper
          .find('div')
          .first()
          .prop('className')
      ).not.toContain(theme.fixed);
    });

    it('uses fixed when set', () => {
      const wrapper = mount(<Tabs fixed theme={theme} />);
      expect(mountToJson(wrapper)).toMatchSnapshot();
    });

    it('does not use inverse by default', () => {
      const wrapper = mount(<Tabs theme={theme} />);
      expect(
        wrapper
          .find('div')
          .first()
          .prop('className')
      ).not.toContain(theme.inverse);
    });

    it('uses inverse when set', () => {
      const wrapper = mount(<Tabs inverse theme={theme} />);
      expect(
        wrapper
          .find('div')
          .first()
          .prop('className')
      ).toContain(theme.inverse);
    });
  });
});
