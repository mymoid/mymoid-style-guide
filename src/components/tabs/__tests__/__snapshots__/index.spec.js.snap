// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tabs #render uses fixed when set 1`] = `
<Tabs
  fixed={true}
  hideMode="unmounted"
  index={0}
  inverse={false}
  theme={Object {}}
>
  <div
    className=""
    data-react-toolbox="tabs"
  >
    <div>
      <nav>
        <span
          className=""
          style={Object {}}
        />
      </nav>
    </div>
  </div>
</Tabs>
`;

exports[`Tabs defaults to only rendering the current tab 1`] = `
<Composition>
  <Tabs
    fixed={false}
    hideMode="unmounted"
    index={0}
    inverse={false}
    theme={
      Object {
        "pointer": "pointer-class",
      }
    }
  >
    <div
      className=""
      data-react-toolbox="tabs"
    >
      <div>
        <nav>
          <span
            className="pointer-class"
            style={Object {}}
          />
        </nav>
      </div>
    </div>
  </Tabs>
</Composition>
`;

exports[`Tabs renders inactive tabs when hideMode is set to display 1`] = `
<Composition
  hideMode="display"
>
  <Tabs
    fixed={false}
    hideMode="display"
    index={0}
    inverse={false}
    theme={
      Object {
        "pointer": "pointer-class",
      }
    }
  >
    <div
      className=""
      data-react-toolbox="tabs"
    >
      <div>
        <nav>
          <span
            className="pointer-class"
            style={Object {}}
          />
        </nav>
      </div>
    </div>
  </Tabs>
</Composition>
`;
